integration-test/src/test/java/org/sagebionetworks/ITDataAccessTest.java

```diff
@@ -861,12 +874,136 @@ public void testGetSubmission() throws SynapseException, JSONObjectAdapterExcept
 				
-				// The first user is still not validated
-				message = assertThrows(SynapseForbiddenException.class, () -> {
-					synapse.getDataAccessSubmission(submissionStatus.getSubmissionId());	
-				}).getMessage();
-				
-				assertEquals("The user must be validated in order to review data access submissions.", message);
+				// The accessor of submission can fetch the submission.
+				Submission submission1 = synapse.getDataAccessSubmission(submissionId);
+				assertEquals(submissionStatus.getSubmissionId(), submission1.getId());
+	}
+
+
+	@Test
+	public void testGetUserAccessApproval() throws SynapseException, JSONObjectAdapterException {
+		SynapseClient synapseTwo = new SynapseClientImpl();
+		userTwoId = SynapseClientHelper.createUser(adminSynapse, synapseTwo, true, true);
+
+		managedAR = new ManagedACTAccessRequirement()
+				.setAccessType(ACCESS_TYPE.DOWNLOAD)
+				.setSubjectIds(Collections.singletonList(new RestrictableObjectDescriptor().setId(project.getId()).setType(RestrictableObjectType.ENTITY)));
+
+		managedAR = adminSynapse.createAccessRequirement(managedAR);
+
+		ResearchProject rp = synapse.getResearchProjectForUpdate(managedAR.getId().toString());
+
+		rp.setInstitution("Sage");
+		rp.setProjectLead("Lead");
+		rp.setIntendedDataUseStatement("intendedDataUseStatement");
+		rp.setAccessRequirementId(managedAR.getId().toString());
+
+		rp = synapse.createOrUpdateResearchProject(rp);
+
+		RequestInterface request = synapse.createOrUpdateRequest(new Request()
+				.setResearchProjectId(rp.getId())
+				.setAccessRequirementId(managedAR.getId().toString())
+				.setAccessorChanges(Arrays.asList(
+						new AccessorChange().setType(AccessType.GAIN_ACCESS).setUserId(adminSynapse.getMyProfile().getOwnerId()),
+						new AccessorChange().setType(AccessType.GAIN_ACCESS).setUserId(synapse.getMyProfile().getOwnerId())
+				)));
+
+		SubmissionStatus submissionStatus = synapse.submitRequest(new CreateSubmissionRequest()
+				.setRequestId(request.getId())
+				.setRequestEtag(request.getEtag())
+				.setSubjectId(project.getId())
+				.setSubjectType(RestrictableObjectType.ENTITY));
+
+		submissionId = submissionStatus.getSubmissionId();
+
+		// Admin is able to fetch the submission
+		Submission submission = adminSynapse.getDataAccessSubmission(submissionStatus.getSubmissionId());
+
+		// create approval for the requirement
+		AccessApproval approval = new AccessApproval();
+		approval.setAccessorId(synapse.getMyProfile().getOwnerId());
+		approval.setRequirementId(managedAR.getId());
+		approval.setRequirementVersion(managedAR.getVersionNumber());
+		approval.setSubmitterId(submission.getSubmittedBy());
+		AccessApproval created = adminSynapse.createAccessApproval(approval);
+
+		String message = assertThrows(SynapseForbiddenException.class, () -> {
+			synapseTwo.getUserAccessApproval(submission.getId());
+		}).getMessage();
+
+		assertEquals("User is not an accessor of the submission " + submission.getId() + ".", message);
```

Comment:
This message is leaking information, just say that you do not have access to this submission